File: app/src/main/java/com/botijasoftware/utils/ui/Button.java
package com.botijasoftware.utils.ui;

import com.botijasoftware.utils.*;


public class Button {
	
	public Button(Rectangle rectangle, Texture imageUp, Texture imageDown) {
	
		rect = rectangle; 
		isPushed = false;
		textureUp = imageUp;
		textureDown = imageDown;
		mVisible = true;
		mSprite = new Sprite( new Rectanglef(rect) , textureUp );
	}
	
	
	
	public void LoadContent(ResourceManager resources) {
        spritebatch = new SpriteBatch(2);
	}
	
	public void Draw() {
		
		if (mVisible) {
            spritebatch.begin();
			spritebatch.DrawSprite(mSprite);
            spritebatch.end();
        }
	}
	
	public boolean isPressed() {
		return isPushed;
	}
	
	public void setPressed() {
		isPushed = true;
		updateImage();
	}
	
	public void setUnpressed() {
		isPushed = false;
		updateImage();
	}
	
	public boolean checkInput(int x, int y) {
		
		if (rect.contains(x, y)) {
			isPushed = !isPushed;
			updateImage();
		}
		return isPushed;
	}
	
	public boolean checkInputNoUnpush(int x, int y) {
		
		if (rect.contains(x, y)) {
			isPushed = true;
			updateImage();
		}
		return isPushed;
	}
	
	private void updateImage() {
		if (isPushed) {
			mSprite.setTexture(textureDown);
		}
		else {
			mSprite.setTexture(textureUp);
		}
	}
	
	public void reloadImages(Texture imageUp, Texture imageDown) {
		textureUp = imageUp;
		textureDown = imageDown;
	}
	
	public void hide() {
		mVisible = false;
	}
	
	public void show() {
		mVisible = true;
	}
	
	public void setVisible(boolean v) {
		mVisible = v;
	}
	
	public boolean isVisible() {
		return mVisible;
	}
	
	public void unPush() {
		isPushed = false;
	}
	
	public void scrollBy(int xScroll, int yScroll) {
		float x = mSprite.getRectangle().X + xScroll;
		float y = mSprite.getRectangle().Y + yScroll;
		mSprite.setPosition(x, y);
		rect.setPosition((int)x, (int)y);
	}
	
	public void moveTo(int x, int y) {
		mSprite.setPosition(x, y);
		rect.setPosition(x, y);
	}
	
	public void moveToX(int x) {
		float y = mSprite.getRectangle().Y;
		mSprite.setPosition(x, y);
		rect.setPosition(x, (int)y);
	}
	
	public void moveToY(int y) {
		float x = mSprite.getRectangle().X;
		mSprite.setPosition(x, y);
		rect.setPosition((int)x, y);
	}
		
	private Rectangle rect;
	private boolean isPushed;
	protected Sprite mSprite;
	protected Texture textureUp;
	protected Texture textureDown;
	protected boolean mVisible;
    protected SpriteBatch spritebatch;



}

File: app/src/main/java/com/botijasoftware/utils/ui/Button.kt
package com.botijasoftware.utils.ui

import com.botijasoftware.utils.*


class Button(private val rect: Rectangle, protected var textureUp: Texture, protected var textureDown: Texture) {

    var isPressed: Boolean = false
        private set
    protected var mSprite: Sprite
    var isVisible: Boolean = false
    protected lateinit var spritebatch: SpriteBatch


    init {
        isPressed = false
        isVisible = true
        mSprite = Sprite(Rectanglef(rect), textureUp)
    }


    fun LoadContent(resources: ResourceManager) {
        spritebatch = SpriteBatch(2)
    }

    fun Draw() {

        if (isVisible) {
            spritebatch.begin()
            spritebatch.DrawSprite(mSprite)
            spritebatch.end()
        }
    }

    fun setPressed() {
        isPressed = true
        updateImage()
    }

    fun setUnpressed() {
        isPressed = false
        updateImage()
    }

    fun checkInput(x: Int, y: Int): Boolean {

        if (rect.contains(x, y)) {
            isPressed = !isPressed
            updateImage()
        }
        return isPressed
    }

    fun checkInputNoUnpush(x: Int, y: Int): Boolean {

        if (rect.contains(x, y)) {
            isPressed = true
            updateImage()
        }
        return isPressed
    }

    private fun updateImage() {
        if (isPressed) {
            mSprite.setTexture(textureDown, false)
        } else {
            mSprite.setTexture(textureUp, false)
        }
    }

    fun reloadImages(imageUp: Texture, imageDown: Texture) {
        textureUp = imageUp
        textureDown = imageDown
    }

    fun hide() {
        isVisible = false
    }

    fun show() {
        isVisible = true
    }

    fun unPush() {
        isPressed = false
    }

    fun scrollBy(xScroll: Int, yScroll: Int) {
        val x = mSprite.rectangle.X + xScroll
        val y = mSprite.rectangle.Y + yScroll
        mSprite.setPosition(x, y)
        rect.setPosition(x.toInt(), y.toInt())
    }

    fun moveTo(x: Int, y: Int) {
        mSprite.setPosition(x.toFloat(), y.toFloat())
        rect.setPosition(x, y)
    }

    fun moveToX(x: Int) {
        val y = mSprite.rectangle.Y
        mSprite.setPosition(x.toFloat(), y)
        rect.setPosition(x, y.toInt())
    }

    fun moveToY(y: Int) {
        val x = mSprite.rectangle.X
        mSprite.setPosition(x, y.toFloat())
        rect.setPosition(x.toInt(), y)
    }


}